connector/src/main/java/org/geysermc/connector/entity/player/PlayerEntity.java

```diff
@@ -102,8 +102,9 @@ public void spawnEntity(GeyserSession session) {
         addPlayerPacket.getMetadata().putAll(metadata);
 
         long linkedEntityId = session.getEntityCache().getCachedPlayerEntityLink(entityId);
-        if (linkedEntityId != -1) {
-            addPlayerPacket.getEntityLinks().add(new EntityLinkData(session.getEntityCache().getEntityByJavaId(linkedEntityId).getGeyserId(), geyserId, EntityLinkData.Type.RIDER, false));
+        Entity linkedEntity = session.getEntityCache().getEntityByJavaId(linkedEntityId);
+        if (linkedEntityId != -1 && linkedEntity != null) {
```

Comment:
Actually, on second thought... can we check for -1 first and then try to grab the Java ID in the entity cache?

```
if (linkedEntityId != -1) {
    Entity linkedEntity = session.getEntityCache().getEntityByJavaId(linkedEntityId);
    if (linkedEntity != null) {
````
```